### PR TITLE
feat: editor

### LINKED ISSUE #56: web: ucdl editor

### PR DESCRIPTION
**Pull request by danieldietzler**
Closes #56

### CODE REVIEW

**Review by Lama64 — APPROVED** [review]

File: server/src/main/java/de/uftos/controller/UcdlEditorController.java [hunk 1]
```diff
@@ -37,8 +38,9 @@ public UcdlEditorController(UcdlEditorService editorService) {
    * @param file the file which contains the new UCDL code.
    * @return a response whether the file could be parsed successfully or not.
    */
-  @PutMapping("/validate")
-  public ParsingResponse validate(@RequestBody MultipartFile file) {
+  @PutMapping(value = "/validate", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+  public ParsingResponse validateUcdlFile(@RequestParam(value = "file") MultipartFile file) {
+    System.out.println(file);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/controller/UcdlEditorController.java:43]: Is this required or a debugging left over?

File: server/src/main/java/de/uftos/services/UcdlEditorService.java [hunk 2]
```diff
@@ -38,6 +38,7 @@ public UcdlEditorService(UcdlRepository ucdlRepository) {
    */
   public ParsingResponse validate(MultipartFile file) {
     try {
+      System.out.println(file);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/UcdlEditorService.java:41]: Same question.

Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/UcdlEditorService.java:41]: Oh yeah I missed those. Good catch!

File: web/src/lib/components/editor.svelte [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      run: onValidate,
+    });
+  });
+</script>
+
+<div class="flex w-full flex-col gap-4">
+  <div class="flex justify-between mr-2 ml-2">
+    <div class="flex gap-4">
+      <Button on:click={() => onInsertTemplate()}>Template einfügen</Button>
+      <Button on:click={() => onValidate()}>Test</Button>
```
[the remark this answers is not in the document]
Comment (marcelfloe) [reply web/src/lib/components/editor.svelte:108]:
```suggestion
      <Button on:click={() => onValidate()}>Code überprüfen</Button>
```

**Pull request merged**